noxfile.py

```diff
@@ -0,0 +1,127 @@
+import nox
+import os
+import sys
+
+
+@nox.session(name="pybamm-requires")
```

Comment:
Ditto!
```suggestion
@nox.session(name="pybamm-requires", reuse_venv=True)
```

Reply:
Note: when I had re-run `pybamm-requires` locally, it failed because `pybind11` was already a directory in the source tree. We might need to wrap something like

```python
if os.path.exists('pybind11_path):
    os.rmdir(pybind11_path)
```

or with 'pathlib.Path.rmdir' or 'shutil.rmtree' as necessary and catch related errors.

Reply:
Note: when I had re-run `pybamm-requires` locally, it failed because `pybind11` was already a directory in the source tree. We might need to wrap something like

```python
if os.path.exists(pybind11_path):
    os.rmdir(pybind11_path)
```

or with 'pathlib.Path.rmdir' or 'shutil.rmtree' as necessary and catch related errors.

Reply:
Note: when I had re-run `pybamm-requires` locally, it failed because `pybind11` was already a directory in the source tree. We might need to wrap something like

```python
if os.path.exists(pybind11_path):
    os.rmdir(pybind11_path)
```

or with `pathlib.Path.rmdir` or `shutil.rmtree` as necessary and catch related errors.

Reply:
Thanks @agriyakhetarpal, looking into it.